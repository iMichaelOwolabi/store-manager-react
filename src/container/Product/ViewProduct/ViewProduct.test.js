import React from 'react';
import { shallow } from 'enzyme';
import { ViewSingleProduct } from '.';

describe('Test the single Products Component', () => {
  const props = {
    product: {
      theProduct: {
        product: {
          productimage: '',
          productname: '',
          price: '',
          quantity: '',
          mininventoryqty: ''
        }
      }
    },
    history: {
      location: {
        pathname: ''
      }
    },
    getProduct: () => {}
  };
  const wrapper = shallow(<ViewSingleProduct {...props} />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the state after change the input value', () => {
    const newValue = 'YoungM';
    wrapper.find('input[type="number"]').simulate('change', {
      target: { name: newValue }
    });
    wrapper.update();
    expect(wrapper.find('input[type="number"]').prop('name')).toContain('quantity');
  });
});
