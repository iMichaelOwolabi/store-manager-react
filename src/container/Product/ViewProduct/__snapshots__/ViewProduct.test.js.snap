// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the single Products Component should match snapshot 1`] = `
<Fragment>
  <AuthNav />
  <div
    className="wrapper"
  >
    <section
      className="productDetails"
    >
      <img
        alt=""
        className="productImage"
        src=""
      />
      <div
        className="details"
      >
        <h1 />
        <p
          className="productInfo"
        >
          Qty:
          <input
            className="quantity"
            name="quantity"
            onChange={[Function]}
            type="number"
            value={1}
          />
        </p>
        <p
          className="productInfo"
        >
          Price:
          <span
            className="moveRight"
          >
            â‚¦
          </span>
        </p>
        <p
          className="productInfo"
        >
          Available Quantity:
          <span
            className="moveRight"
          />
        </p>
        <p
          className="productInfo"
        >
          Minimum Qty:
          <span
            className="moveRight"
          />
        </p>
        <button
          className="productBtn"
          type="submit"
        >
          Add to Cart
        </button>
        <button
          className="productBtn"
          type="submit"
        >
          Modify
        </button>
        <button
          className="deleteButton"
          type="submit"
        >
          Delete
        </button>
      </div>
    </section>
  </div>
</Fragment>
`;
